ste/sender-blockBlobFromLocal.go

```diff
@@ -92,7 +92,7 @@ func (u *blockBlobUploader) generatePutWholeBlob(id common.ChunkID, blockIndex i
 		jptm.LogChunkStatus(id, common.EWaitReason.Body())
 		var err error
 		if jptm.Info().SourceSize == 0 {
-			_, err = u.destBlockBlobURL.Upload(jptm.Context(), bytes.NewReader(nil), u.headersToApply, u.metadataToApply, azblob.BlobAccessConditions{})
+			_, err = u.destBlockBlobURL.Upload(jptm.Context(), bytes.NewReader(nil), u.headersToApply, u.metadataToApply, azblob.BlobAccessConditions{}, u.destBlobTier)
```

Comment:
The verification and logging performed by `AttemptSetBlobTier()` should be performed here. There are restrictions on what tier can be applied to a blob. `u.dstBlobTier` should be set if the tier clears those conditions.  Also, the log messages wrt. to `SetTier` failures should be preserved.